deepspeed/runtime/zero/partition_parameters.py

```diff
@@ -789,24 +789,21 @@ class Init(InsertPostInitMethodToModuleSubClasses):
     apply_param_persistence = False
     override_module_apply = get_config_default(DeepSpeedZeroConfig, "override_module_apply")
 
-    def __init__(
-        self,
-        module=None,
-        data_parallel_group=None,
-        mem_efficient_linear=True,
-        remote_device=None,
-        pin_memory=False,
-        config_dict_or_path=None,
-        config=None,
-        enabled=True,
-        dtype=None,
-        mpu=None,
-        zero_param_parallel_group=None,
-        zero_quantized_weights=False,
-        zero_quantized_nontrainable_weights=False,
-        sequence_data_parallel_group=None,
-        param_swapper=None,
-    ):
+    def __init__(self,
+                 module=None,
+                 data_parallel_group=None,
+                 mem_efficient_linear=True,
+                 remote_device=None,
+                 pin_memory=False,
+                 config_dict_or_path=None,
+                 config=None,
+                 enabled=True,
+                 dtype=None,
+                 mpu=None,
+                 zero_param_parallel_group=None,
+                 zero_quantized_weights=False,
+                 zero_quantized_nontrainable_weights=False,
```

Comment:
```suggestion
                 zero_quantized_nontrainable_weights=False,
                 sequence_data_parallel_group=None,
```
Let's properly deprecate this over the next few versions. Below, in the `__init__` method, do something like:
```python
if sequence_data_parallel_group is not None:
    print_rank_0("WARNING: 'sequence_data_parallel_group' is deprecated and will be removed. Use 'data_parallel_group' instead.", force=True)
    if data_parallel_group is not None:
        raise ValueError("Both 'data_parallel_group' and 'sequence_data_parallel_group' were specified. Please provide only one of these arguments.")
    data_parallel_group = sequence_data_parallel_group
```